forms/gridfield/GridFieldExportButton.php

```diff
@@ -165,7 +164,14 @@ public function generateExportFileData($gridField) {
 			}
 		}
 
-		return $fileData;
+		// Convert the $fileData array into csv by capturing fputcsv's output
+		ob_start();
+		$fp = fopen('php://output', 'w');
+		foreach($fileData as $line) {
+			fputcsv($fp, $line, $this->csvSeparator, $this->csvEnclosure);
+		}
+		fclose($fp);
+		return ob_get_clean();
```

Comment:
I would suggest using a temp buffer instead of piggy backing off the output buffer (which could already contain buffered content).

http://www.php.net/manual/en/function.fputcsv.php#74118